external/audiocodec/streaming/player.c

```diff
@@ -29,615 +29,555 @@
-    const int32_t kMaxBytesChecked = 8*1024;//128 * 1024;
-    uint8_t buf[kMaxReadBytes];
-    ssize_t bytesToRead = kMaxReadBytes;
-    ssize_t totalBytesRead = 0;
-    ssize_t remainingBytes = 0;
-    bool reachEOS = false;
-    uint8_t *tmp = buf;
-
-    do {
-        if (pos >= *inout_pos + kMaxBytesChecked)
-            break;	// Don't scan forever.
-
-        if (remainingBytes < 4)
-		{
-            if (reachEOS)
-                break;
-
-            memcpy(buf, tmp, remainingBytes);
-            bytesToRead = kMaxReadBytes - remainingBytes;
-
-            /*
-             * The next read position should start from the end of
-             * the last buffer, and thus should include the remaining
-             * bytes in the buffer.
-             */
-            totalBytesRead = _source_read_at(fp, pos + remainingBytes,
-                                         buf + remainingBytes, bytesToRead);
-
-            if (totalBytesRead <= 0)
-                break;
-
-            reachEOS = (totalBytesRead != bytesToRead);
-            remainingBytes += totalBytesRead;
-            tmp = buf;
-            continue;
-        }
-
-        uint32_t header = _u32_at(tmp);
-
-        if (match_header != 0 && (header & MP3_FRAME_HEADER_MASK) != (match_header & MP3_FRAME_HEADER_MASK))
-		{
-            ++pos;
-            ++tmp;
-            --remainingBytes;
-            continue;
-        }
-
-        size_t frame_size;
-        uint32_t sample_rate, num_channels, bitrate;
-        if (!_parse_header(header, &frame_size, &sample_rate, &num_channels, &bitrate, NULL))
-		{
-            ++pos;
-            ++tmp;
-            --remainingBytes;
-            continue;
-        }
-
-        // We found what looks like a valid frame,
-        // now find its successors.
-
-        ssize_t test_pos = pos + frame_size;
-
-        valid = true;
-        const int FRAME_MATCH_REQUIRED = 2;
-        for (int j = 0; j < FRAME_MATCH_REQUIRED; ++j)
-		{
-            uint8_t temp[4];
-            ssize_t retval = _source_read_at(fp, test_pos, temp, sizeof(temp));
-            if (retval < (ssize_t)sizeof(temp))
-			{
-                valid = false;
-                break;
-            }
-
-            uint32_t test_header = _u32_at(temp);
-
-            if ((test_header & MP3_FRAME_HEADER_MASK) != (header & MP3_FRAME_HEADER_MASK))
-			{
-                valid = false;
-                break;
-            }
-
-            size_t test_frame_size;
-            if (!_parse_header(test_header, &test_frame_size, NULL, NULL, NULL, NULL))
-			{
-                valid = false;
-                break;
-            }
-
-            test_pos += test_frame_size;
-        }
-
-        if (valid)
-		{
-            *inout_pos = pos;
-
-            if (out_header != NULL)
-                *out_header = header;
+	if (*inout_pos == 0) {
+		// Skip an optional ID3 header if syncing at the very beginning of the datasource.
+		for (;;) {
+			uint8_t id3header[10];
```

Comment:
As possible as you can, please avoid use constant value to enhance readability.
#define MP3_AAU_HEADER_SIZE 10